### PR TITLE
Allow specifying filepath to only run dumps from that file

### PR DESCRIPTION
**Pull request by julial-pvep**
Partially resolves https://github.com/Pioneer-Valley-Books/rails-response-dumper/issues/56

(line number to come in later PR)

### CODE REVIEW

File: lib/rails_response_dumper/runner.rb [hunk 1]
```diff
@@ -13,10 +13,11 @@ def initialize(options)
 
     def run_dumps
       dumps_dir = Rails.root.join('dumps')
+      file_path = options[:filename] || 'dumpers/**/*.rb'
```
Comment (jdufresne) [inline lib/rails_response_dumper/runner.rb:16]: Maybe name this "glob" as it looks like glob paths are supported.

File: lib/rails_response_dumper/runner.rb [hunk 2]
```diff
@@ -13,10 +13,11 @@ def initialize(options)
...
       FileUtils.rm_rf dumps_dir
       FileUtils.mkdir_p dumps_dir
 
-      Dir[Rails.root.join('dumpers/**/*.rb')].each { |f| require f }
+      Dir[Rails.root.join(file_path)].each { |f| require f }
```
Comment (jdufresne) [inline lib/rails_response_dumper/runner.rb:20]: Hmm, this doesn't filter by the passed in filename. If a dumper has a `require` on another dumper, that 2nd dumper will still run. This just limits what will be required, not what will run.

**Review by jdufresne — COMMENTED** [review]
Very nice!

Could you add a note to the changelog, please?

File: lib/rails_response_dumper/runner.rb (hunk 2 above)
Comment (julial-pvep) [reply lib/rails_response_dumper/runner.rb:20]: Hi @jdufresne , just coming back to this. Was wondering if you could give an example of a case where one might require one dumper from another dumper's file and we expect it to not run?

I keep thinking about this like rspec shared examples, where I assume we'd want to run the shared tests required in the specified file.

File: lib/rails_response_dumper/option_parser.rb [hunk 3]
```diff
@@ -7,6 +7,16 @@ def self.parse_options!
     options = {}
 
     OptionParser.new do |opts|
+      opts.banner += ' [files or directories]'
```
Comment (jdufresne) [inline lib/rails_response_dumper/option_parser.rb:10]: Does specifying a directory work?

Might to useful to emphasize that this is a glob.

```suggestion
      opts.banner += ' [glob]'
```

File: spec/cli_spec.rb [hunk 4]
```diff
@@ -120,12 +120,104 @@
...
         expect(stdout).not_to include(invalid_number_of_statuses)
         expect(stdout).not_to include(dumper_2_invalid_status_code)
         expect(status.exitstatus).to eq(1)
       end
     end
   end
+
+  context 'with filename argument' do
+    context 'without previous dumps' do
+      before { FileUtils.rm_rf("#{APP_DIR}/dumps") }
```
Comment (jdufresne) [inline spec/cli_spec.rb:134]: Maybe the output should go to a temporary directory so tests are isolated.

**Review by jdufresne — APPROVED** [review]
Nice! Thoughts on documenting this in the README too?

File: spec/cli_spec.rb (hunk 4 above)
Comment (jdufresne) [reply spec/cli_spec.rb:134]: > Or that each test has its own output directory?

This one.

I would expect this directory would be created with something like `mktmpdir` and cleaned up after the example runs.

https://ruby-doc.org/3.2.1/stdlibs/tmpdir/Dir.html#method-c-mktmpdir

Using a separate directory per tests allows the tests to run in parallel.

File: README.md [hunk 5]
```diff
@@ -120,6 +120,15 @@ ResponseDumper.define 'Users' do
 end
 ```
 
+## Running Dumps for Specific Files
```
Comment (jdufresne) [inline README.md:123]: Up to you, but I think this should go directly under "usage" as it seems morefundamental than before/after hooks.

File: CHANGELOG.md [hunk 6]
```diff
@@ -9,6 +9,8 @@ headers from the dumper output.
 - The dump property `response.status_text` was renamed to `response.statusText`
   to match the JavaScript `Response` constructor.
 
+- Add option to specify dump file(s) to run from CLI as an alternative to running all dumps.
```
Comment (jdufresne) [inline CHANGELOG.md:12]:
```suggestion
- Add option to specify dump files to run from CLI as an alternative to running all dumps.
```

File: spec/cli_spec.rb [hunk 7]
```diff
@@ -127,12 +127,105 @@
...
+          expect(status.exitstatus).to eq(0)
+        end
+      end
+
+      context 'with changes to the dump file' do
+        before do
+          FileUtils.mv(
+            "#{APP_DIR}/dumpers/tests_response_dumper.rb",
+            "#{APP_DIR}/configurable_dumpers/initial_tests_response_dumper.rb"
+          )
```
Comment (jdufresne) [inline spec/cli_spec.rb:195]: What is going on here? I don't think we should be modifying the repo files directly. If we need to changes files mid-test, we should be working out of a temporary directory.

**Review by jdufresne — APPROVED** [review]

File: spec/cli_spec.rb (hunk 7 above)
Comment (julial-pvep) [reply spec/cli_spec.rb]: A couple things occurred to me when going to do this:

1) This is the PR in which we make the dumper directory configurable, so no need for an early PR 🤦 
2) We don't need to actually move the files to test the new file content, we can just reference the altered version of the file directly from configurable_dumps.

Let me know what you think: [Reference configurable directory directly](https://github.com/Pioneer-Valley-Books/rails-response-dumper/pull/63/commits/61212e1b803c8fa8acafe5c634eaa2602a868f88)

**Review by jdufresne — APPROVED** [review]

**Review by giverm — APPROVED** [review]

**Pull request merged**